### PR TITLE
feat(core): Add credential runtime checks and prevent tampering in manual run

### PR DESCRIPTION
**Pull request by ivov**

### CODE REVIEW

**Review by krynble — CHANGES_REQUESTED** [review]
Great work. I think the only necessary change was the one I've mentioned in Linear, where we need to make sure the current user did not tamper the request by changing existing nodes with credentials they don't have access to, neither adding credentials they shouldn't access.

File: packages/cli/src/UserManagement/PermissionChecker.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		// if disallowed, flag only first node using first inaccessible cred
+
+		const nodeToFlag = credIdsToNodes[inaccessibleCredIds[0]][0];
+
+		throw new NodeOperationError(nodeToFlag, 'Node has no access to credential', {
+			description: 'Please recreate the credential or ask its owner to share it with you.',
+		});
+	}
+
+	private static mapCredIdsToNodes(workflow: Workflow) {
```
Comment (krynble) [inline packages/cli/src/UserManagement/PermissionChecker.ts:55]: I loved the creation of this function, this is very useful in so many places.

**Review by krynble — DISMISSED** [review]
LGTM

**Review by krynble — DISMISSED** [review]

**Review by krynble — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.203.0`

**Pull request merged**